### PR TITLE
Added basic page object pattern implementation for Windows apps

### PR DESCRIPTION
**Pull request by jamesmcroft**
## PR summary

This PR is the starting point for creating automated UI tests with Appium for Windows applications following the page object pattern. 

Additionally, there are some starting point `WindowsElement` wrappers contained within this PR which were initially created to test functions of core controls in the Windows 10 SDKs. Note, these are not a complete set of wrappers and will be included in a separate piece of work.

Sample test projects have been added for the Windows Alarms & Clock application from my original [Windows UWP Appium page object pattern example](https://github.com/jamesmcroft/uwp-appium-pop-example), and another sample project added for the XAML Controls Gallery application.

## PR checklist

- [x] Sample tests have been added/updated and pass
- [x] Code styling has been run on all new source file changes
- [x] Contains **NO** breaking changes

## References

- [Windows (UWP) Appium Page Object Pattern Example](https://github.com/jamesmcroft/uwp-appium-pop-example)

### CODE REVIEW

File: samples/WindowsAlarmsAndClock/Pages/AlarmPage.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    using Legerity.Pages;
+    using Legerity.Windows.Extensions;
+
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+    using OpenQA.Selenium;
+    using OpenQA.Selenium.Appium;
+
+    /// <summary>
+    /// Defines the alarm page of the Windows Alarms & Clock application.
```
Comment (tom-made) [inline samples/WindowsAlarmsAndClock/Pages/AlarmPage.cs:16]: Can you fix this invalid XML? Believe this should be `&amp;`

File: samples/WindowsAlarmsAndClock/Pages/EditAlarmPage.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+    using Legerity.Pages;
+    using Legerity.Windows.Elements.Core;
+    using Legerity.Windows.Extensions;
+
+    using OpenQA.Selenium;
+    using OpenQA.Selenium.Appium.Windows;
+
+    /// <summary>
+    /// Defines the add/edit alarm page of the Windows Alarms & Clock application.
```
Comment (tom-made) [inline samples/WindowsAlarmsAndClock/Pages/EditAlarmPage.cs:13]: Same as previous comment, this should be `&amp;`

File: samples/WindowsAlarmsAndClock/Pages/AlarmPage.cs (hunk 1 above)
Comment (jamesmcroft) [reply samples/WindowsAlarmsAndClock/Pages/AlarmPage.cs]: This should be fixed now

File: samples/WindowsAlarmsAndClock/Pages/EditAlarmPage.cs (hunk 2 above)
Comment (jamesmcroft) [reply samples/WindowsAlarmsAndClock/Pages/EditAlarmPage.cs]: Sorted :) (as above)

**Comment by tom-made** [comment]
Let's start fleshing out features for the rest of the work before carrying on. Help to track where we're going with this.

**Pull request merged**